src/contract/aci.ts

```diff
@@ -0,0 +1,487 @@
+      }
+      const txOpt = {
+        ...opt,
+        gasLimit: opt.gasLimit ?? GAS_MAX,
+        callData,
+        nonce: opt.nonce ??
+          (opt.top != null &&
+            parseInt((await this.getAccount(callerId, { hash: opt.top })).nonce) + 1)
+      }
+      const tx = fn === 'init'
+        ? (await this.contractCreateTx({ ...txOpt, code: instance.bytecode, ownerId: callerId })).tx
+        : await this.contractCallTx({ ...txOpt, callerId, contractId })
+
+      const { callObj, ...dryRunOther } = await this.txDryRun(tx, callerId, opt)
+      await handleCallError(callObj, tx)
+      res = { ...dryRunOther, tx: TxObject({ tx }), result: callObj }
+    } else {
+      const tx = await this.contractCallTx({
+        ...opt,
+        gasLimit: opt.gasLimit ?? await instance._estimateGas(fn, params, opt),
+        callerId,
+        contractId,
+        callData
+      })
+      res = await sendAndProcess(tx, opt)
+    }
+    if (opt.callStatic === true || res.txData.blockHeight != null) {
+      res.decodedResult = fnACI.returns != null && fnACI.returns !== 'unit' && fn !== 'init' &&
+        instance.calldata.decode(instance._name, fn, res.result.returnValue)
+      res.decodedEvents = instance.decodeEvents(res.result.log, opt)
+    }
+    return res
+  }
+
+  /**
+  * @param address Contract address that emitted event
+  * @param nameHash Hash of emitted event name
+  * @param options Options
+  * @returns Contract name
+  * @throws {MissingEventDefinitionError}
+  * @throws {AmbiguousEventDefinitionError}
+  */
+  function getContractNameByEvent (
+    address: string,
+    nameHash: BigInt,
+    { contractAddressToName }:
+    {contractAddressToName?: {[key: string]: string}}): string {
+    const addressToName = { ...instance.options.contractAddressToName, ...contractAddressToName }
+    if (addressToName[address] != null) return addressToName[address]
+
+    const matchedEvents = [
+      instance._aci.encodedAci,
+      ...instance._aci.externalEncodedAci
+    ]
+      .filter(({ contract }) => contract?.event)
+      .map(({ contract }) => [contract.name, contract.event.variant])
+      .map(([name, events]) => events.map((event: {}) => [name, Object.keys(event)[0]]))
+      .flat()
+      .filter(([, eventName]) => BigInt('0x' + hash(eventName).toString('hex')) === nameHash)
+    switch (matchedEvents.length) {
+      case 0: throw new MissingEventDefinitionError(nameHash.toString(), address)
+      case 1: return matchedEvents[0][0]
+      default: throw new AmbiguousEventDefinitionError(address, matchedEvents)
+    }
+  }
+
+  /**
+  * Decode Events
+  * @alias module:@aeternity/aepp-sdk/es/contract/aci
+  * @param events Array of encoded events (callRes.result.log)
+  * @param options Options
+  * @returns DecodedEvents
+  */
+  instance.decodeEvents = (
+    events: Event[],
+    { omitUnknown, ...opt }: {omitUnknown?: boolean} = {}
+  ): Array<DecodedEvent | null> => events
```

Comment:
Actually, there can't be null in result because of the last line

Reply:
[`if ((omitUnknown ?? false) && error instanceof MissingEventDefinitionError) return null`
](https://github.com/aeternity/aepp-sdk-js/pull/1511/files#diff-26f6070ca63b8ca32e57bf46ae6c5b2aa050adedb71b463467f4fe1bc183828bR440)returns null

Reply:
it is removed in `.filter(e => e)`